designer/server/src/main/scala/pl/touk/nussknacker/ui/process/processingtype/provider/ReloadableProcessingTypeDataProvider.scala

```diff
@@ -18,56 +19,53 @@ import pl.touk.nussknacker.ui.security.api.NussknackerInternalUser
  * Another thing that needs careful consideration is handling exception during ProcessingTypeData creation/closing - probably during
  * close we want to catch exception and try to proceed, but during creation it can be a bit tricky...
  */
-class ReloadableProcessingTypeDataProvider(
-    loadMethod: IO[ProcessingTypeDataState[ProcessingTypeData, CombinedProcessingTypeData]]
-) extends ProcessingTypeDataProvider[ProcessingTypeData, CombinedProcessingTypeData]
+class ReloadableProcessingTypeDataProvider[Data <: AutoCloseable, CombinedData](
+    loadMethod: IO[ProcessingTypeDataState[Data, CombinedData]]
+) extends ProcessingTypeDataProvider[Data, CombinedData](ProcessingTypeDataState.uninitialized)
     with LazyLogging {
 
-  // We initiate state with dumb value instead of calling loadMethod() to avoid problems with dependency injection
-  // cycle - see NusskanckerDefaultAppRouter.create
-  private var stateValue: ProcessingTypeDataState[ProcessingTypeData, CombinedProcessingTypeData] = emptyState
-
-  override private[processingtype] def state
-      : ProcessingTypeDataState[ProcessingTypeData, CombinedProcessingTypeData] = {
-    synchronized {
-      stateValue
-    }
-  }
-
-  def reloadAll(): IO[Unit] = synchronized {
```

Comment:
This `synchronized` stopped working after we changed implementation to `IO`-based